nucypher/cli/commands/deploy.py

```diff
@@ -364,14 +400,25 @@ def upgrade(general_config, actor_options, retarget, target_address, ignore_depl
             raise click.BadArgumentUsage(message="--target-address is required when using --retarget")
         if not actor_options.force:
             click.confirm(CONFIRM_RETARGET.format(contract_name=contract_name, target_address=target_address), abort=True)
-        receipt = ADMINISTRATOR.retarget_proxy(contract_name=contract_name,target_address=target_address)
+        receipt = ADMINISTRATOR.retarget_proxy(contract_name=contract_name,target_address=target_address, confirmations=confirmations)
         message = SUCCESSFUL_RETARGET.format(contract_name=contract_name, target_address=target_address)
         emitter.message(message, color='green')
         paint_receipt_summary(emitter=emitter, receipt=receipt)
+
     else:
         if not actor_options.force:
+            # Check for human verification of versioned upgrade details
             click.confirm(CONFIRM_BEGIN_UPGRADE.format(contract_name=contract_name), abort=True)
-        receipts = ADMINISTRATOR.upgrade_contract(contract_name=contract_name, ignore_deployed=ignore_deployed)
+            if deployer._ownable:  # Only ownable + upgradeable contracts apply
+                old_agent = ContractAgency.get_agent(agent_class=deployer.agency, registry=github_registry)
+                new_agent = ContractAgency.get_agent(agent_class=deployer.agency, registry=local_registry)
```

Comment:
Nice use of two agents with different registries! I think it's the first time I see this used that explicitly and I'm liking it a lot :)

Reply:
Looks good, but I think it's over complicated considering that we only want to get version of contracts, maybe change it to `registry.search`

Reply:
@vzotova - Here's how that ends up looking:

```python
old_contract = github_registry.search(contract_name=contract_name)[-1]  # latest GH version
new_contract = local_registry.search(contract_name=contract_name)[-1]   # latest local version
```

Reply:
:heavy_check_mark: